test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/matchers/ResponseHeadersMatcher.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.services.test.utils.core.matchers;
+
+import static java.util.Optional.empty;
+
+import java.util.Optional;
+
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.collection.IsMapContaining;
+
+import uk.gov.justice.services.test.utils.core.http.ResponseData;
+
+public class ResponseHeadersMatcher extends ResponseMatcher<ResponseData> {
+	private Optional<IsMapContaining<String, Object>> headersMatcher = empty();
+
+	public static ResponseHeadersMatcher headers() {
+		return new ResponseHeadersMatcher();
+	}
+
+	public ResponseHeadersMatcher hasHeader(final Matcher<String> keyMatcher, final Matcher<Object> valueMatcher) {
+		this.headersMatcher = Optional.of(new IsMapContaining<>(keyMatcher, valueMatcher));
+		return this;
+	}
+
+	@Override
+	protected boolean matchesSafely(ResponseData item, Description mismatchDescription) {
+		final MultivaluedMap<String, Object> actualHeaders = item.getHeaders();
+		
+		if (headersMatcher.isPresent() && headersMatcher.get().matches(actualHeaders)) {
+			return true;
+
+		} else {
+			describeMismatch(headersMatcher.get(), actualHeaders, mismatchDescription);
+			return false;
+		}
+
+	}
+
+	private void describeMismatch(IsMapContaining<String, Object> isMapContaining,
+			MultivaluedMap<String, Object> actualHeaders, Description mismatchDescription) {
+		mismatchDescription.appendText("Headers ");
+		isMapContaining.describeMismatch(actualHeaders, mismatchDescription);
+
+	}
+
+	@Override
+	public void describeTo(Description description) {
+		headersMatcher.ifPresent(matcher -> description.appendText("Headers ").appendDescriptionOf(matcher));
```

Comment:
Is this Headers output a duplicate? if above also?
I can't remember exactly how the matcher description works and it's not obvious from code review screen :)